.github/workflows/python_lint.yml

```diff
@@ -28,20 +26,68 @@ jobs:
           - "3.11"
     steps:
       - uses: actions/checkout@v3
-
       - name: Setup Micromamba
         uses: mamba-org/provision-with-micromamba@main
         with:
           python_version: ${{ matrix.python_version }}
           cache-downloads: true
           cache-env: true
           environment-file: ./environment.yml
-
       - name: Run black
         run: |
           black --check python
 
-      # Include `--format=github` to enable automatic inline annotations.
+  ruff:
+    name: Ruff Python ${{ matrix.python_version }} - ${{ matrix.os }} - ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+        python_version:
+          - "3.11"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          python_version: ${{ matrix.python_version }}
+          cache-downloads: true
+          cache-env: true
+          environment-file: ./environment.yml
       - name: Run ruff
         run: |
           ruff check --format=github python
+
+
+  mypy:
+    name: Mypy Python ${{ matrix.python_version }} - ${{ matrix.os }} - ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+        python_version:
+          - "3.11"
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Micromamba
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          python_version: ${{ matrix.python_version }}
+          cache-downloads: true
+          cache-env: true
+          environment-file: ./environment.yml
+      - name: Run mypy
+        run: |
+          mypy --ignore-missing-imports python/ribasim
```

Comment:
Should this run only on the ribasim subdirectory? Also add newline at the end.

Reply:
Unfortunately, it is hard to do that locally because of this: https://github.com/pypa/setuptools/issues/3518
However, we don't hit that on the CI so I've added it.